sklearn/inspection/partial_dependence.py

```diff
@@ -345,22 +352,30 @@ def partial_dependence(estimator, X, features, response_method='auto',
                 "With the 'recursion' method, the response_method must be "
                 "'decision_function'. Got {}.".format(response_method)
             )
+        msg = "'estimator' parameter must be a fitted estimator"
+        if isinstance(estimator, BaseGradientBoosting):
+            fitted_attribute = 'estimators_'
+        else:
+            fitted_attribute = 'n_iter_'
+        check_is_fitted(estimator, fitted_attribute, msg=msg)
```

Comment:
That wasn't caught before?

Reply:
it was caught too late when computing the partial dependence if I recall properly

Reply:
The tests still pass removing this line. I don't understand why this is needed.

Reply:
Uhm true, maybe linked to https://github.com/scikit-learn/scikit-learn/commit/da0bdab4ac6841533e136c6ba24dd3bc0bb75188 where I was getting a mixed of NotFittedError and ValueError.
We should probably use NotFittedError in fact